### PR TITLE
database: add Instructions and code for day 2

### PR DESCRIPTION
**Pull request by Soypete**
Day two of the workshop is databases + web apps.
The examples are how to do databases in go. There is also the code for the goose/sqlc live demo.

updates #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,8 @@ This repo contains the examples and exercises for The O'reilly online learning c
 # Databases for webservices
 
 ## TODO: 
-* Exercise 1: Connect to hosted db with dumby data
+* Exercise 1: Connect to a live database
+* live coding: sqlc
 * Exercise 2: Add client and interface
 * Exercise 3: Unit tests with mock client
 * Example: integration testing with github actions
```
Comment (Soypete) [inline README.md:26]: add a todo here. This is not currently being covered in the course and I do not have any examples of this

File: database/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Day 2 Databases and your Go app
+
+This section of the course contains examples and exercises for effectively integrating databases to your Web app. The following are examples of finished exercises for day 2 of the [O'reilly Media Online Learning Course](https://www.oreilly.com/live-events/go-for-web-development-in-3-weeks/0636920091015/). They can be completed by following along with the instruction or independently.
```
Comment (Soypete) [inline database/README.md:3]:
```suggestion
This section of the course contains examples and exercises for effectively leveraging a database in the backend of your Web services.. The following are examples of finished exercises for day 2 of the [O'reilly Media Online Learning Course](https://www.oreilly.com/live-events/go-for-web-development-in-3-weeks/0636920091015/). They can be completed by following along with the instruction or independently.
```

File: database/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## Exercise 1
+
+Use the appropriate database driver and go module to connect to the provided postgres database. The username and password are provided on the slide. Once you have connected run a query based on the following schema...
```
Comment (Soypete) [inline database/README.md:7]:
```suggestion
Use the appropriate database driver and go module to connect to the provided postgres database. The username and password are provided by the instructor (See below if completing independently). Once you have connected run a query based on the following schema...
```

File: database/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Questions:
+
+* what are the differences between manually creating a database object vs generating one with sqlc?
+
+## Exercise 3 mock database
+
+Using your new database interface mock the database functions into your [tests from last week.](../restful-go/ex-4-tests/framework_test.go)
+
+TODO: copy tests from cli-game. and edit the interface to be inside the test sweet.
```
Comment (Soypete) [inline database/README.md:36]:
```suggestion
TODO: copy tests from cli-game. and edit the interface to be inside the test suite.
```

File: restful-go/ex-2-webframeworks/framework.go [hunk 5]
```diff
@@ -46,3 +47,21 @@ func deleteUsername(w http.ResponseWriter, r *http.Request) {
...
+	r := chi.NewRouter()
+	r.Use(middleware.RequestID)
+	r.Use(middleware.Logger)
+	r.Use(middleware.Recoverer)
+
+	r.Route("/register", func(r chi.Router) {
+		// subroutes for register
+		r.Route("/{username}", func(r chi.Router) {
+			r.Get("/get", getUsername)          // GET /register/123/get
+			r.Get("/update", updateUsername)    // PUT /register/123/update // TODO: this is a get because I am not providing a body
```
Comment (Soypete) [inline restful-go/ex-2-webframeworks/framework.go:61]: remove comment

File: restful-go/ex-2-webframeworks/framework.go [hunk 6]
```diff
@@ -46,3 +47,21 @@ func deleteUsername(w http.ResponseWriter, r *http.Request) {
...
+			r.Delete("/delete", deleteUsername) // DELETE /register/123/delete //TODO: should this have /delete in the path?
```
Comment (Soypete) [inline restful-go/ex-2-webframeworks/framework.go:62]: remove comment

File: restful-go/ex-2-webframeworks/framework.go (hunk 5 above)
Comment (Soypete) [reply restful-go/ex-2-webframeworks/framework.go:61]: will do in future work. #14

File: restful-go/ex-2-webframeworks/framework.go (hunk 6 above)
Comment (Soypete) [reply restful-go/ex-2-webframeworks/framework.go:62]: will do in future work. #14

File: database/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+Use the appropriate database driver and go module to connect to the provided postgres database. The username and password are provided by the instructor (See below if completing independently). Once you have connected run a query based on the following schema...
+
+TODO: include an ERD or other schema docs.
+
+_NOTE_: If you are completeling this indepently. You can run postgres locally in a docker container using the following command.
+
+```
+docker pull postgres
+docker run -e POSTGRES_PASSWORD=postgres -e POSTGRES_USERNAME=postgres -p 5431:5432 postgres
```
Comment (todpunk) [inline database/README.md:15]: use not-postgres for username/password so they aren't accidentally trapped in being correct by default.

File: go.sum [hunk 8]
```diff
@@ -1,2 +1,9 @@
 github.com/go-chi/chi v1.5.4 h1:QHdzF2szwjqVV4wmByUnTcsbIg7UGaQ0tPF2t5GcAIs=
 github.com/go-chi/chi v1.5.4/go.mod h1:uaf8YgoFazUOkPBG7fxPftUylNumIev9awIWOENIuEg=
+github.com/go-sql-driver/mysql v1.6.0/go.mod h1:DCzpHaOWr8IXmIStZouvnhqoel9Qv2LBy8hT2VhHyBg=
```
Comment (todpunk) [inline go.sum:3]: why is mysql here?

**Review by todpunk — COMMENTED** [review]
Directionally sound so far, I've added a couple nits for a course audience that I think aids in learning (or avoiding confusion in the case of the mysql lib). Feel free to ignore those nits.

File: go.sum (hunk 8 above)
Comment (Soypete) [reply go.sum:4]: there is an example my sql script for the slides. it is an artifact of other prs. it should be fixed once they are merged

**Pull request merged**